sympy/functions/special/gamma_functions.py

```diff
@@ -799,11 +799,30 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
 
     def fdiff(self, argindex=2):
         if argindex == 2:
-            n, z = self.args[:2]
+            n, z = self.args
             return polygamma(n + 1, z)
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        s, z = self.args
+        d = z.subs(x, 0) # expansion point
+        if d is S.NaN:
+            return S.NaN
```

Comment:
It is not necessary to always return `nan`. `z` may be well behaved and have a valid limit value at `x = 0`. Consider for example `z = sin(x)/x`.